### PR TITLE
[W5.7][W17-3]Koh Zheng Wei

### PR DESCRIPTION
**Pull request by Zenious**
added enhancement to addressbook-level2
-Ability to see the total number of Person in address book without having to go through verbose output generated by the command `list`

### CODE REVIEW

File: src/seedu/addressbook/parser/Parser.java [hunk 1]
```diff
@@ -11,17 +11,7 @@
...
-import seedu.addressbook.commands.Command;
-import seedu.addressbook.commands.DeleteCommand;
-import seedu.addressbook.commands.ExitCommand;
-import seedu.addressbook.commands.FindCommand;
-import seedu.addressbook.commands.HelpCommand;
-import seedu.addressbook.commands.IncorrectCommand;
-import seedu.addressbook.commands.ListCommand;
-import seedu.addressbook.commands.ViewAllCommand;
-import seedu.addressbook.commands.ViewCommand;
+import seedu.addressbook.commands.*;
```
Comment (PakornUe) [inline src/seedu/addressbook/parser/Parser.java]: importing .* is unsafe

File: src/seedu/addressbook/commands/StatsCommand.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package seedu.addressbook.commands;
+
+public class StatsCommand extends Command {
```
Comment (PakornUe) [inline src/seedu/addressbook/commands/StatsCommand.java]: Shouldn't "count" be more intuitive than "stats" in this context?

File: test/input.txt [hunk 3]
```diff
@@ -140,6 +140,14 @@
...
   list
 
+##########################################################
+# test stats command
+##########################################################
+
+  # show stats
+  stats
+  list
+
```
Comment (PakornUe) [inline test/input.txt:150]: maybe add more test cases? (ex. clear then run stat for one more time)

File: src/seedu/addressbook/commands/StatsCommand.java [hunk 4]
```diff
@@ -13,7 +17,8 @@
 
     @Override
     public CommandResult execute() {
-        int size = addressBook.size();
+        List<ReadOnlyPerson> allPersons = addressBook.getAllPersons().immutableListView();
```
Comment (PakornUe) [inline src/seedu/addressbook/commands/StatsCommand.java]: This seems like a pretty good idea.

**Review by PakornUe — CHANGES_REQUESTED** [review]
Most of the code and documentation is of good quality.

Tests are mostly adequate except the I/O one which could be more rigorous.

However, importing .* is haphazard and should be fixed.

**Review by PakornUe — APPROVED** [review]

**Review by PakornUe — APPROVED** [review]

**Review by PakornUe — APPROVED** [review]
Ok, everything seems fine

File: src/seedu/addressbook/Main.java [hunk 5]
```diff
@@ -112,8 +112,9 @@ private CommandResult executeCommand(Command command)  {
             storage.save(addressBook);
             return result;
         } catch (Exception e) {
-            ui.showToUser(e.getMessage());
-            throw new RuntimeException(e);
+//            ui.showToUser(e.getMessage());
+//            throw new RuntimeException(e);
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/Main.java:116]: Don't comment out code. Go ahead and delete all unused code. We have version control precisely to revert at any point.

File: src/seedu/addressbook/commands/StatsCommand.java [hunk 6]
```diff
@@ -0,0 +1,25 @@
+package seedu.addressbook.commands;
+
+import seedu.addressbook.data.person.ReadOnlyPerson;
+
+import java.util.List;
+
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/StatsCommand.java:6]: Write descriptive header comments for all public classes/methods.

File: src/seedu/addressbook/commands/StatsCommand.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+public class StatsCommand extends Command {
+
+
```
Comment (parvathy-sudhir-pillai) [inline src/seedu/addressbook/commands/StatsCommand.java:9]: Avoid stray line breaks.

File: test/java/seedu/addressbook/commands/StatsCommandTest.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import org.junit.Test;
+import seedu.addressbook.data.AddressBook;
+import seedu.addressbook.data.person.ReadOnlyPerson;
+import seedu.addressbook.util.TestUtil;
+import seedu.addressbook.util.TypicalPersons;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
```
Comment (parvathy-sudhir-pillai) [inline test/java/seedu/addressbook/commands/StatsCommandTest.java:14]: Add descriptive comments for public classes.

**Review by parvathy-sudhir-pillai — APPROVED** [review]
Good effort on this PR @Zenious. In particular, good job on,

updating I/O tests
updating JUnit tests
keeping the PR relatively clean
updating user guide

Suggestions to improve further:
update developer guide 

As before, please close the PR after viewing comments.

**Pull request closed without merging**